src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/AbstractBuildEventValueProvider.cs

```diff
@@ -30,7 +32,22 @@ internal abstract partial class AbstractBuildEventValueProvider : InterceptingPr
             using (ProjectLockReleaser access = await _projectLockService.ReadLockAsync())
             {
                 Microsoft.Build.Construction.ProjectRootElement projectXml = await access.GetProjectXmlAsync(_unconfiguredProject.FullPath).ConfigureAwait(true);
-                return await _helper.GetPropertyAsync(projectXml, defaultProperties).ConfigureAwait(true);
+                string rawValue = await _helper.GetPropertyAsync(projectXml, defaultProperties).ConfigureAwait(true);
+                if (rawValue == null) return rawValue;
+
+                Regex regex = new Regex("%([0-9A-Fa-f][0-9A-Fa-f])");
```

Comment:
Seems like any escape logic we add should go in `AbstractBuildEventHelper` instead of this class.